File: 031-Missing Number in an Array/MissingNumberinanArray.cpp
#include<iostream>
#include<vector>
using namespace std;

int findMissingNumber(vector<int>& arr, int n){
    int total = n * (n + 1)/ 2;

    int sum = 0;
    for(int i = 0; i < arr.size(); i++){
        sum += arr[i];
    }
    return total - sum;
}
int main(){
    vector<int> arr = {1, 2, 4, 5, 6};
    int n = 6;
    cout<<"Missing Number: "<<findMissingNumber(arr,n)<<endl;
    return 0;
}

File: 031-Missing Number in an Array/MissingNumberinanArray.py
def find_missing_number(arr, n):
    total = n * (n + 1) // 2  # Expected sum from 1 to n
    return total - sum(arr)  # Missing number is the difference

# Example usage
arr = [1, 2, 4, 5, 6]
n = 6
print("Missing Number:", find_missing_number(arr, n))